app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt

```diff
@@ -0,0 +1,161 @@
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
+import eu.davidea.flexibleadapter.FlexibleAdapter
+import eu.davidea.flexibleadapter.Payload
+import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.databinding.ClearDatabaseControllerBinding
+import eu.kanade.tachiyomi.ui.base.controller.DialogController
+import eu.kanade.tachiyomi.ui.base.controller.FabController
+import eu.kanade.tachiyomi.ui.base.controller.NucleusController
+import eu.kanade.tachiyomi.util.system.toast
+import eu.kanade.tachiyomi.util.view.doOnApplyWindowInsetsCompat
+import eu.kanade.tachiyomi.util.view.liftAppbarWith
+import eu.kanade.tachiyomi.util.view.marginBottom
+
+class ClearDatabaseController :
+    NucleusController<ClearDatabaseControllerBinding, ClearDatabasePresenter>(),
+    FlexibleAdapter.OnItemClickListener,
+    FlexibleAdapter.OnUpdateListener,
+    FabController {
+
+    private var recycler: RecyclerView? = null
+    private var adapter: FlexibleAdapter<ClearDatabaseSourceItem>? = null
+
+    private var menu: Menu? = null
+
+    private var actionFab: ExtendedFloatingActionButton? = null
+    private var actionFabScrollListener: RecyclerView.OnScrollListener? = null
+
+    init {
+        setHasOptionsMenu(true)
+    }
+
+    override fun createBinding(inflater: LayoutInflater): ClearDatabaseControllerBinding {
+        return ClearDatabaseControllerBinding.inflate(inflater)
+    }
+
+    override fun createPresenter(): ClearDatabasePresenter {
+        return ClearDatabasePresenter()
+    }
+
+    override fun getTitle(): String? {
+        return activity?.getString(R.string.clear_database)
+    }
+
+    override fun onViewCreated(view: View) {
+        super.onViewCreated(view)
+        liftAppbarWith(binding.recycler, true)
+
+        adapter = FlexibleAdapter<ClearDatabaseSourceItem>(null, this, true)
+        binding.recycler.adapter = adapter
+        binding.recycler.layoutManager = LinearLayoutManager(view.context)
+        binding.recycler.setHasFixedSize(true)
+        adapter?.fastScroller = binding.fastScroller
+        recycler = binding.recycler
+        val fabBaseMarginBottom = binding.fab.marginBottom
+        binding.recycler.doOnApplyWindowInsetsCompat { v, insets, _ ->
+
+            binding.fab.updateLayoutParams<ViewGroup.MarginLayoutParams> {
+                bottomMargin = fabBaseMarginBottom + insets.getInsets(WindowInsetsCompat.Type.systemBars()).bottom
+            }
+            v.post {
+                // offset the binding.recycler by the binding.fab's inset + some inset on top
+                v.updatePaddingRelative(
+                    bottom = insets.getInsets(WindowInsetsCompat.Type.systemBars()).bottom + binding.fab.marginBottom +
+                        (binding.fab.height)
+                )
+            }
+        }
```

Comment:
The following method handles insets among other things for the recycler
```suggestion
        scrollViewWith(recycler, true, afterInsets = { insets ->
                binding.fab.updateLayoutParams<ViewGroup.MarginLayoutParams> {
                    bottomMargin = insets.getInsets(systemBars()).bottom + fabBaseMarginBottom 
                }
            }
        )
```

Reply:
had to `recycler!!` as well